src/compiler/sys.ts

```diff
@@ -297,6 +304,131 @@ namespace ts {
                 };
             }
 
+            function createWatchedFileSet() {
+                const dirWatchers = createFileMap<DirWatcher>();
+                const recursiveDirWatchers = createFileMap<DirWatcher>();
+                const fileWatcherCallbacks = createFileMap<FileWatcherCallback>();

```

Comment:
Ah it wouldn't work now. Maybe a file needs a list of callbacks. Will update later.